config/rollup.config.js

```diff
@@ -0,0 +1,73 @@
+import path from 'path';
+
+import makePlugins from './plugins/index';
+import makeDevServerPlugins from './plugins/server';
+
+const isProduction = process.env.NODE_ENV === 'production';
+
+// default config that is used across all builds
+const config = {
+  input: 'index.js',
+  treeshake: {
+    // We assume reading a property of an object never has side-effects so we
+    // disable this option as it can significantly reduce bundle size but
+    // can potentially break functionality if you rely on getters or errors
+    // from illegal property access.
+    propertyReadSideEffects: false
+  }
+};
+
+export default function makeConfig(commandOptions) {
+  const builds = [];
+
+  // iife build (for use with dev-server)
+  const iife = {
+    ...config,
+    plugins: [
+      ...makePlugins(isProduction),
+      // if rollup has been ran with `--open`, include the dev server plugins
+      ...(commandOptions.open ? makeDevServerPlugins(commandOptions) : [])
+    ],
+    output: {
+      format: 'iife',
+      name: 'Spectacle',
+      file: path.join('dist', 'spectacle.js')
+    }
+  };
+
+  // UMD build
+  const umd = {
+    ...config,
+    external: ['react', 'react-is', 'react-dom', 'prop-types'],
+    plugins: [...makePlugins(isProduction)],
+    output: [
+      {
+        name: 'Spectacle',
+        sourcemap: false,
+        file: `./dist/umd/spectacle.js`,
```

Comment:
This won't be published per https://github.com/FormidableLabs/spectacle/blob/task/rewrite/.npmignore (needs a `!` whitelist entry if we like this name).

Overall, I think you probably want to do a full build then `npm pack` and then view the tgz contents to make sure that everything you see in the tarball is correct. Also, we can remove `es|lib` directories as if we take this PR we won't produce babel transpiled files anymore.

Reply:
I was intending to have the different build formats seperated out by directory within the `dist` directory - i.e. `dist/es/spectacle.js`, `/dist/cjs/spectacle.js`, `dist/umd/spectacle` (and the relevant fields in `package.json` pointing to these locations) - should we remove `dist` from the npmignore (but keep ignoring source maps)?

Reply:
You can adjust the `.npmignore` to match what currently happens on `master`, namely:

1. Ignore the example bundle for the demo app
2. Include the production and development consumable entry points (UMD, CJS, ESM, etc.)